wooey/tasks.py

```diff
@@ -69,9 +94,30 @@ def submit_script(**kwargs):
     job.status = WooeyJob.RUNNING
     job.save()
 
+    # Use lists so they can be passed into the threads and appended in place
+    # we only extract + join in the main thread so no issues
+    stdout, stderr = [], []
+
     proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=abscwd)
 
-    stdout, stderr = proc.communicate()
+    # We need to use threads to capture the IO, otherwise they will block one another
+    # i.e. a check against stderr will sit waiting on stderr before returning
+    tout = Thread(target=append_output, args=(proc.stdout, stdout))
+    tout.start()
+
+    terr = Thread(target=append_output, args=(proc.stderr, stderr))
+    terr.start()
+
+    prev_std = None
+
+    # Loop until the process is complete + both stdout/stderr have EOFd
+    while proc.poll() is None or tout.is_alive() or terr.is_alive():
+        # Only update if there is output from the script
+        if stdout + stderr != prev_std:
```

Comment:
I think it would be better to use a tuple for this comparison so we remove any possible issues with concatenation